agent/sensoryOrgans/ear/web/src/index.tsx

```diff
@@ -0,0 +1,138 @@
+
+  source.connect(analyser);
+  draw();
+
+  function draw() {
+    const WIDTH = canvas.width
+    const HEIGHT = canvas.height;
+    requestAnimationFrame(draw);
+    analyser.getByteTimeDomainData(dataArray);
+    canvasCtx.fillStyle = 'rgb(200, 200, 200)';
+    canvasCtx.fillRect(0, 0, WIDTH, HEIGHT);
+    canvasCtx.lineWidth = 2;
+    canvasCtx.strokeStyle = 'rgb(0, 0, 0)';
+    canvasCtx.beginPath();
+    let sliceWidth = WIDTH * 1.0 / bufferLength;
+    let x = 0;
+    for (let i = 0; i < bufferLength; i++) {
+      let v = dataArray[i] / 128.0;
+      let y = v * HEIGHT / 2;
+      if (i === 0) {
+        canvasCtx.moveTo(x, y);
+      } else {
+        canvasCtx.lineTo(x, y);
+      }
+      x += sliceWidth;
+    }
+    canvasCtx.lineTo(canvas.width, canvas.height / 2);
+    canvasCtx.stroke();
+  }
+}
+
+const postAudioData = async (blob: Blob): Promise<string> => {
+  const myHeaders = new Headers();
+  myHeaders.append("Accept", "application/json");
+  myHeaders.append("Accept", "text/plain");
+  myHeaders.append("Accept", "*/*");
+
+  const formData = new FormData();
+  formData.append("audio_data", blob, 'temp_recording');
+  const response = await fetch("http://0.0.0.0:5000/transcribe", {
+    method: "POST",
+    body:
+      formData,
+    headers: myHeaders
+  });
+  let responseText = '';
+  if (response.status === 200) {
+    responseText = await response.text();
+  }
+  console.log(responseText);
+  return responseText;
+}
+
+
+const generateClipElement = (blob: Blob) => {
+  const audioURL = window.URL.createObjectURL(blob);
+  const clipContainer = document.createElement('article');
+  const audio = document.createElement('audio');
+  clipContainer.classList.add('clip');
+  audio.setAttribute('controls', '');
+  audio.controls = true;
+  audio.src = audioURL;
+  clipContainer.appendChild(audio);
+  soundClips.appendChild(clipContainer);
+  console.log(clipContainer)
+}
+
+const main = async () => {
+  const constraints = { audio: true };
+  let chunks: any[] = [];
+
+  const stream = await navigator.mediaDevices.getUserMedia(constraints);
+  const mediaRecorder = new MediaRecorder(stream);
+  visualize(stream);
+
+  mediaRecorder.ondataavailable = async (e) => {
+    const currentBlob = e.data;
+    console.log(currentBlob);
+    if (currentBlob.size > 0) {
+      chunks.push(currentBlob);
+      const blob = new Blob(chunks, { 'type': 'audio/ogg; codecs=opus' });
+      generateClipElement(blob);
+      const transcribedText = await postAudioData(blob);
+      if (chunks.length > TARGET_MAX_INFERENCE / INTERVAL) {
+        chunks = [];
+        currentText = [...currentText, [transcribedText]];
+        mediaRecorder.stop();
+        mediaRecorder.start();
+      } else {
+        // add the transcription to the chunk at the end of the array
+        const currentCurrentText = currentText[currentText.length - 1]
+        currentCurrentText.push(transcribedText)
+      }
+    }
+  }
+  mediaRecorder.start(INTERVAL);
```

Comment:
this is wrong. i'll figure it out
it kinda works though
![image](https://user-images.githubusercontent.com/10282244/215293820-67a06aee-5c8d-4aa7-b454-22053aa94e6f.png)